### PR TITLE
Make llama2 run on GPU using SPMD.

### PR DESCRIPTION
**Pull request by vanbasten23**
# What does this PR do?

Make Llama2 run on GPU with SPMD on PyTorch/XLA.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

cc: @miladm

### CODE REVIEW

**Review by yeounoh — APPROVED** [review]
LGTM thank you @vanbasten23 ,let's wait for @jonb377 's approval as well.

File: src/transformers/training_args.py
Comment (jonb377) [inline src/transformers/training_args.py:1422]: Does this need to go into HF main?

**Review by jonb377 — APPROVED** [review]
Awesome! Thanks Xiongfei, looking forward to seeing some benchmarks 🚀

File: src/transformers/training_args.py
Comment (vanbasten23) [reply src/transformers/training_args.py:1422]: Yeah, this should. Let me create another PR in HF main.

**Comment by vanbasten23** [comment]
Thanks for the review!

**Pull request merged**
